app/components/Settings/Settings.jsx

```diff
@@ -175,8 +176,16 @@ class Settings extends React.Component {
         });
     }
 
+    _handleSettingsEntryChange(setting, input) {
+        if (!input.target) {
+            this._onChangeSetting(setting, {target: {value: input}});
+        } else {
+            this._onChangeSetting(setting, input);
+        }
+    }
+
     _onChangeSetting(setting, e) {
-        e.preventDefault();
+        if (e.preventDefault) e.preventDefault();
```

Comment:
Function _onChangeSettings called in two places, one of which is callback for input, and the other is simple function. So when it called from simple function, no event (e) is provided as a parameter, and if you'd try to call e.preventDefault() without check, it will cause an error

Reply:
Function _onChangeSettings called in two places, one of which is callback for input and the other is simple function. So when it called from simple function, no event (e) is provided as an argument, and if you'd try to call e.preventDefault() without check, it will cause an error